### ISSUE TITLE
Normalize handling of BlockStatement brackets at FunctionBody node

### ISSUE THREAD

**Issue by cgewecke** [ConsenSys/solidity-parser#58]
[`FunctionBody`](https://github.com/ConsenSys/solidity-parser/blob/master/solidity.pegjs#L1355)'s type is `BlockStatement` but its parent consumes the enclosing brackets resulting in anomalous start / end values for the node. This is an issue for the Solium linter which wants to be able to analyze all `BlockStatement`s in the same way. 

The [`FunctionDeclaration`](https://github.com/ConsenSys/solidity-parser/blob/master/solidity.pegjs#L1261) rule was inherited from pegjs and it's not obvious why they chose to implement things this way. From the linter's perspective it puts relevant information in the wrong place. 

@duaraghav8's solparse has revised the relevant rules so they work for his context [here](https://github.com/duaraghav8/solparse/blob/master/solidity-parser.pegjs#L1723) and [here](https://github.com/duaraghav8/solparse/blob/master/solidity-parser.pegjs#L1633).

**Comment by duaraghav8** [comment]
Thanks for opening this @cgewecke 
Me and cgewecke discussed why I did this [here](https://github.com/duaraghav8/solparse/issues/15) (8. Start/end Position).
So my fix was based on my interpretation of how the parser **should have** worked. And then I wrote Solium's code to comply with solparse's way of working. And now its heavily dependant on how solparse parses ```FunctionDeclaration``` & ```BlockStatement```

Once we decide how the parser should finally work, I can change solium's code (if need be), but I think my version is how it should be.

Opinions?

ps- @raineorshine since you implemented the start / end position logic, maybe you too would like to give your input.
